gui/default/index.html

```diff
@@ -585,12 +585,16 @@ <h4 class="panel-title">
                       </tr>
                       <tr>
                         <th><span class="fas fa-fw fa-share-alt"></span>&nbsp;<span translate>Shared With</span></th>
-                        <td class="text-right no-overflow-ellipse word-break-all">
+                        <td class="text-right no-overflow-ellipse overflow-break-word">
                           <span ng-repeat="device in otherDevices(folder.devices)">
-                            <span ng-switch="completion[device.deviceID][folder.id].remoteState">
-                              <span ng-switch-when="notSharing" data-original-title="{{'The remote device has not accepted sharing this folder.' | translate}}" tooltip>{{deviceName(devices[device.deviceID])}}<sup>1</sup><span ng-if="!$last">,</span></span>
-                              <span ng-switch-when="paused" data-original-title="{{'The remote device has paused this folder.' | translate}}" tooltip>{{deviceName(devices[device.deviceID])}}<sup>2</sup><span ng-if="!$last">,</span></span>
-                              <span ng-switch-default>{{deviceName(devices[device.deviceID])}}<span ng-if="!$last">,</span></span>
+                            <span ng-if="folder.type !== 'receiveencrypted' && device.encryptionPassword" class="text-nowrap">
+                              <span class="fa fa-lock"></span>&nbsp;<!-- Avoid stray space...
+                         --></span><!-- Avoid stray space...
+                         --><span ng-switch="completion[device.deviceID][folder.id].remoteState"><!-- Avoid stray space...
+                           --><span ng-switch-when="notSharing" data-original-title="{{'The remote device has not accepted sharing this folder.' | translate}}" tooltip>{{deviceName(devices[device.deviceID])}}<sup>1</sup></span><!-- Avoid stray space...
+                           --><span ng-switch-when="paused" data-original-title="{{'The remote device has paused this folder.' | translate}}" tooltip>{{deviceName(devices[device.deviceID])}}<sup>2</sup></span><!-- Avoid stray space...
+                           --><span ng-switch-default>{{deviceName(devices[device.deviceID])}}</span><!-- Avoid stray space...
```

Comment:
optional:  
This adds a unrelated noise to the diff of the functional change - ideally these space-avoidance-comme\nts (and the tab to space indentation change somewhere above) would be their own PR/commit.

Reply:
Will look into it and make a separate PR if it can be split off sensibly.

Reply:
This PR was rebased on top of #9438, implementing the desired separation.  Besides the commit history, the changes are identical.